### PR TITLE
Add Example for IQM Resonance

### LINKED ISSUE #128: Update user guide to reflect Resonance usage
The user guide is mainly focussed towards on-prem devices, however, with Resonance, one just needs an API Token that can be added to the backend. This should be reflected in the docs as well.

### PR DESCRIPTION
**Pull request by manzanillo**
The Hello World is now split into a section for the on-prem devices using Cortex CLU and IQM Resonance.
The example from the Resonance docs is also added.

This fixes #128

### CODE REVIEW

File: docs/user_guide.rst [hunk 1]
```diff
@@ -8,12 +8,29 @@ code snippets and check the output yourself.
...
+6. If you're connecting to a real quantum computer, the output should show almost half of the measurements resulting in '00000' and almost half in '11111' – if this is the case, things are set up correctly!
+
+You can find a video guide on how to set things up, `here <https://www.iqmacademy.com/tutorials/resonance/>`.
+More ready-to-run examples can also be found on `IQM Academy <https://www.iqmacademy.com/tutorials/>` .
+
+
+Hello, world! (On-prem devices)
 -------------
 
 Here's the quickest and easiest way to run a small computation on an IQM quantum computer and check that
```
Comment (freetonik) [inline docs/user_guide.rst:36]: Perhaps here it's worth explaining that the instructions below apply to a non-cloud case (in contrast to the new paragraph above).

File: docs/user_guide.rst [hunk 2]
```diff
@@ -8,12 +8,29 @@ code snippets and check the output yourself.
...
+
+Here's the quickest and easiest way to run a small computation on an IQM quantum computer and check that
+things are set up correctly:
+
+1. Login to `IQM Resonance <https://resonance.meetiqm.com>` with your credentials
+2. Generate an API token using the dashboard
+3. Download one of the demo notebooks on `IQM Academy <https://www.iqmacademy.com/tutorials/>` or the `resonance_example.py example file <https://raw.githubusercontent.com/iqm-finland/qiskit-on-iqm/main/src/iqm/qiskit_iqm/examples/resonance_example.py>`_ (Save Page As...)
+4. Install Qiskit on IQM as instructed below
+5. Add your API token to the example (this can be done as a parameter `token` to the `IQMProvider` constructor or by setting the environment variable `IQM_TOKEN`)
+5. Run the Jupyter notebook (or run ``$ python resonance_example.py`` if you decided to go for the Python script)
```
Comment (freetonik) [inline docs/user_guide.rst:26]:
```suggestion
5. Run the Jupyter notebook (or run ``python resonance_example.py`` if you decided to go for the Python script)
```

File: src/iqm/qiskit_iqm/examples/resonance_example.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+qc_transpiled = transpile_to_IQM(qc, backend)
+print(qc_transpiled.draw(output="text"))
+
+# Run circuit
+job = backend.run(qc_transpiled, shots=SHOTS)
+print(job.result().get_counts())
+
+# Plot result
+result_dict = job.result().get_counts()
+visualization.plot_histogram(result_dict)
```
Comment (freetonik) [inline src/iqm/qiskit_iqm/examples/resonance_example.py:33]: If the user runs this in a terminal, as instructed, the visualization won't render, right?

File: docs/user_guide.rst [hunk 4]
```diff
@@ -8,12 +8,29 @@ code snippets and check the output yourself.
...
 
-Hello, world!
+Hello, world! (IQM Resonance)
+-------------
+
+Here's the quickest and easiest way to run a small computation on an IQM quantum computer and check that
+things are set up correctly:
+
+1. Login to `IQM Resonance <https://resonance.meetiqm.com>` with your credentials
+2. Generate an API token using the dashboard
```
Comment (kukushechkin) [inline docs/user_guide.rst:22]: I would suggest adding some suggestions on how to handle the access token, maybe just copy the test from Resonance docs. I can imaging how people generate it in this step, close the popup, then go to step 5 and like "where is the access token?", have seen it in the Hanken hackathon few weeks ago.

**Comment by freetonik** [comment]
If you want to publish the updated docs [here](https://iqm-finland.github.io/qiskit-on-iqm/), then please include a changelog update with a minor version bump. This way a new version will be released, and docs will be re-built and re-published.

File: src/iqm/qiskit_iqm/examples/resonance_example.py (hunk 3 above)
Comment (manzanillo) [reply src/iqm/qiskit_iqm/examples/resonance_example.py:33]: It should open up a Python window that displays it, but let's just remove it, it is not important anyway

File: docs/user_guide.rst [hunk 5]
```diff
@@ -8,22 +8,39 @@ code snippets and check the output yourself.
 
 .. note::
 
-   At the moment IQM does not provide a quantum computing service open to the general public.
-   Please contact our `sales team <https://www.meetiqm.com/contact-us/>`_ to set up your access to an IQM quantum
-   computer.
```
Comment (smite) [inline docs/user_guide.rst:13]: So no sales team/contact link is needed anymore?

File: docs/user_guide.rst [hunk 6]
```diff
@@ -8,22 +8,39 @@ code snippets and check the output yourself.
...
+4. Install Qiskit on IQM as instructed below.
+5. Add your API token to the example (this can be done as a parameter `token` to the `IQMProvider` constructor or by setting the environment variable `IQM_TOKEN`)
+5. Run the Jupyter notebook (or run ``python resonance_example.py`` if you decided to go for the Python script).
+6. If you're connecting to a real quantum computer, the output should show almost half of the measurements resulting in '00000' and almost half in '11111' – if this is the case, things are set up correctly!
+
+You can find a video guide on how to set things up, `here <https://www.iqmacademy.com/tutorials/resonance/>`.
+More ready-to-run examples can also be found on `IQM Academy <https://www.iqmacademy.com/tutorials/>` .
+
+
+Hello, world! (On-prem devices)
```
Comment (smite) [inline docs/user_guide.rst:33]:
```suggestion
Hello, world! (On-premises devices)
```

File: docs/user_guide.rst (hunk 5 above)
Comment (manzanillo) [reply docs/user_guide.rst:13]: Yes, as there is Resonance, one can learn about Resonance on the main website

**Review by kukushechkin — APPROVED** [review]

**Review by smite — APPROVED** [review]

**Pull request merged**